homeassistant/components/soma/__init__.py

```diff
@@ -120,4 +121,25 @@ async def async_update(self):
             self.is_available = False
             return
         self.current_position = 100 - response["position"]
+        try:
+            response = await self.hass.async_add_executor_job(
+                self.api.get_battery_level, self.device["mac"]
+            )
+        except RequestException:
+            _LOGGER.error("Connection to SOMA Connect failed")
+            self.is_available = False
+            return
+        if response["result"] != "success":
+            _LOGGER.error(
+                "Unable to reach device %s (%s)", self.device["name"], response["msg"]
+            )
+            self.is_available = False
+            return
+        battery = 2.5 * response["battery_level"] - 922
+        if battery < 0:
+            battery = 0
+        elif battery > 100:
+            battery = 100
+        self.state_attrs = {}
```

Comment:
Initialize the dict in the init function.

Reply:
done